src/utilities/_hypre_utilities.h

```diff
@@ -231,27 +231,37 @@ void hypre_error_handler(const char *filename, HYPRE_Int line, HYPRE_Int ierr, c
 #if defined(HYPRE_DEBUG)
 /* host assert */
 #define hypre_assert(EX) do { if (!(EX)) { fprintf(stderr, "[%s, %d] hypre_assert failed: %s\n", __FILE__, __LINE__, #EX); hypre_error(1); assert(0); } } while (0)
+
+/* host assert and exit */
+#define hypre_assert_exit(comm, EX) do { if (!(EX)) { hypre_assert(EX); hypre_MPI_Abort(comm, 1); } } while (0)
```

Comment:
Since `hypre_assert` calls `assert(0)` which exits, then it's too late to call `MPI_Abort` right?